### ISSUE TITLE
Will there be training code for the dVAE?

### ISSUE THREAD

**Issue by FomalhautB** [openai/DALL-E#8]
If we want to train the discrete VAE, do we have to tune it by our self or we can wait till the code release?

**Comment by adityaramesh** [comment]
All of the details for the training procedure are described in Appendix A of our paper. I might add an example training script if there's a lot of demand, but otherwise I'm happy to answer any questions that are not already answered by the paper.

**Comment by FomalhautB** [comment]
Are there maybe some hyperparameters that are not given in the paper? for example, how many cycles do the cosine annealing do? Do you use cosine annealing with a restart or without restart? The paper says "The KL weight β is increased from 0 to 6.6 over the first 5000 updates", do you use cosine annealing after the 5000 updates? Sorry for so many questions, but it is a little bit hard to find out all the hyperparameters.

**Comment by adityaramesh** [comment]
The KL term is fixed to 6.6 after the 5000 updates -- I believe this trick was first popularized by this paper: https://arxiv.org/abs/1511.06349

Thanks for pointing out the ambiguity -- I'll edit the paper to make this clearer. Please feel free to ask if you have any further questions.

**Comment by airsplay** [comment]
> All of the details for the training procedure are described in Appendix A of our paper. I might add an example training script if there's a lot of demand, but otherwise I'm happy to answer any questions that are not already answered by the paper.

Thanks for releasing the code/model code and answering all the detailed questions here. According to the paper, there is actually a huge improvement over the previous VQVAE training method. It would definitely be a great help if an example training script could be added.

**Comment by adityaramesh** [comment]
The improvement over VQVAE actually isn't huge, and we don't make this claim anywhere. It's mainly that the relaxation is simpler and avoids the need for tricks like the EMA loss, dead code revival, etc. and scales up to larger vocabulary sizes.

I'll consider adding a training script to the repo (my original code is in TensorFlow and I'd rather port to PyTorch first).

**Comment by neel04** [comment]
@adityaramesh thanks for such a wonderful repo! I would really appreciate if you could provide a training script for the `dVAE` because there seems to be a huge improvement over vanilla VQ-VAE (compressibility is `256x256 -> 32x32, where in the vanilla model 128x128 -> 32x32`).

Is this 4x improvement because of much more lossier compression, or does dVAE do something special not mentioned? 🤔 

In all scenarios, training code is much, much appreciated!!! 🤗

**Comment by rashindrie** [comment]
+1 for dVAE training script!

**Comment by jaylnne** [comment]
also want training script 🙏
